### ISSUE TITLE
creating a new user-pass for a software existing already in the database overwrites it 

### ISSUE THREAD

**Issue by Benjamin-Pasternak** [pranavtailor/CS419-PasswordVault#4]
If a user creates another entry that has the same software name as a previous entry, the username and password will be overwritten. So if an adversary gets the master password or access in some other way such as by user error, they can overwrite all of the user's passwords. 

Additionally, in a user context; the user cannot save alt account info. Like for example social media where one person might have more than one account.

**Comment by pranavtailor** [comment]
Hello,

To address your concerns:
1.  We purposely made our software able to 'overwrite' credentials. How else would a user be able to update their username/password when he changes them? When you say an adversary can overwrite a users passwords, why would someone want to do that? They gain absolutely nothing. Theoretically, an adversary would like to see the credentials for their own malicious intent. Simply changing passwords is pointless. Therefore, we believe this is NOT a bug and the software functions exactly how we want it to.
2.  In terms of a user not able to save alternate account info, this is also NOT a bug, and is more so just something we felt was unnecessary. It is quite rare for someone to have the exact same software name string, with different credentials. Pertaining to your social media example, if a user wanted to do this, it would probably be in their best interest to differentiate the software name slightly based on account. Otherwise, how else would the user know which credentials are for which account?

To summarize, we strongly believe neither of these are bugs, so we dispute your claims.

Regards,
Pranav

**Comment by symxmyz233** [comment]
**It is not a bug.**
